client/asset/eth/eth.go

```diff
@@ -218,7 +224,16 @@ func (d *Driver) DecodeCoinID(coinID []byte) (string, error) {
 			return "", fmt.Errorf("invalid hex address %q", coinID)
 		}
 		return common.HexToAddress(hexAddr).String(), nil
+	case len(coinIDTakerFoundMakerRedemption) + common.AddressLength*2 + 2:
+		hexCoinID := string(coinID)
+		if !strings.HasPrefix(hexCoinID, coinIDTakerFoundMakerRedemption) {
+			return "", fmt.Errorf("coinID %q has no %s prefix", coinID, coinIDTakerFoundMakerRedemption)
+		}
+		coinIDCopy := make([]byte, len(hexCoinID))
+		copy(coinIDCopy, hexCoinID)
+		return string(coinIDCopy), nil
```

Comment:
Looks like `DecodeCoinID` strives to return a copy of its argument, so it won't be modified explicitely ? Doing same here.

Reply:
I think the weirdness you see is because the go-ethereum package will capitalize some hex for some reason that I forget atm. @martonp what was it again?

I do not that performing `string(hexCoinID)` would alter the slice anyway.

Reply:
Oh, this was it https://medium.com/@danielque/why-ethereum-addresses-have-some-capitalized-letters-b8e55728ecaa

Reply:
Yeah, [looks like](https://go.dev/play/p/k8JpRvQLAda) wrapping `[]byte` with `string()` protects underlying slice from implicit modification,

so, no need to do deep-copy, thanks, simplified!